### PR TITLE
SensitivityAnalysis - use "by path" methods

### LINKED ISSUE #269: SensitivityAnalysis - use "by path" methods

### PR DESCRIPTION
**Pull request by PavelBal**
Fixes #269

### CODE REVIEW

**Comment by PavelBal** [comment]
This PR avoids creating `Parameter` objects and uses the `byPath` methods implemented in ospsuite.

**Review by IndrajeetPatil — CHANGES_REQUESTED** [review]

**Comment by IndrajeetPatil** [comment]
Hmm, are _ospsuite_ binaries not up to date?

```r
== Failed tests ================================================================
  -- Error (test-sensitivity-calculation.R:26:1): (code run outside of `test_that()`) --
  Error: 'getQuantityValuesByPath' is not an exported object from 'namespace:ospsuite'
  Backtrace:
      x
   1. \-esqlabsR::sensitivityCalculation(...) at test-sensitivity-calculation.R:26:0
  
  [ FAIL 1 | WARN 6 | SKIP 0 | PASS 114 ]
```

We are fetching them from here: https://ci.appveyor.com/api/projects/open-systems-pharmacology-ci/ospsuite-r/artifacts/ospsuite.zip

**Comment by PavelBal** [comment]
Damn, this is due to https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1017. Will probably have to wait until Michael triggers a PK-Sim build....

File: R/sensitivityCalculation.R [hunk 1]
```diff
@@ -104,16 +104,20 @@ sensitivityCalculation <- function(simulation,
...
-    param <- getParameter(parameterPath, simulation)
-
-    if (param$isConstant) {
-      constantParamPaths <- c(constantParamPaths, parameterPath)
-    } else {
+    # Check if the parameter is given by an explicit formula
+    if (ospsuite::isExplicitFormulaByPath(
+      path = parameterPath,
+      simulation = simulation
+    )) {
```
[the remark this answers is not in the document]
Comment (PavelBal) [reply R/sensitivityCalculation.R]: OK done.

**Review by IndrajeetPatil — APPROVED** [review]

**Comment by IndrajeetPatil** [comment]
I am looking into the failing example.

**Comment by PavelBal** [comment]
This is strange, why does it work locally?

Unlocking means that the user could add new fields AFTER creation of the object, which is not good. "Cannot add bindings" implies that `defaultPlotConfiguration$legendCaptionSize` is not present in the class, but it actually is... It also works locally.

**Comment by PavelBal** [comment]
@IndrajeetPatil Did you remove the field `legendCaptionSize` from default plot configuration at some point?

**Comment by IndrajeetPatil** [comment]
> @IndrajeetPatil Did you remove the field legendCaptionSize from default plot configuration at some point?

Yepp, it has been renamed. Already made a commit to fix this. I think that should fix the issue.

**Pull request merged**